Let operating system choose port

ChaosCoder opened the pull request:
When opening a unix socket, the system chooses a port for you, if you do not specify one (`port = 0`). This PR leverages this behaviour if you specify the port `0` on starting the server and returns the used port.

This PR adds new test cases and updates the documentation of the `Server.start(port)` method.

This fixes #27 and addresses the problem stated in #20 in a different but safer way.

KaneCheshire commented on Sources/Peasy/Internal/Socket.swift:
```diff
@@ -30,9 +30,26 @@ final class Socket {
 		address.sin6_addr = .localhost
 		let size = socklen_t(MemoryLayout<sockaddr_in6>.size)
-		let success = withUnsafePointer(to: &address) { $0.withMemoryRebound(to: sockaddr.self, capacity: Int(size)) { Darwin.bind(tag, $0, size) >= 0 } }
+		let success = withUnsafePointer(to: &address) { address -> Bool in
```
Can we use type inference here to remove the `-> Bool` and also the `return` from L34? Not a big deal if not, just a bit tidier

ChaosCoder replied:
Done

KaneCheshire commented on Sources/Peasy/Internal/Socket.swift:
```diff
@@ -30,9 +30,26 @@ final class Socket {
+					Darwin.getsockname(tag, $0, usedAddressSize)
+				}
+			}
+		}
```
Man these Darwin APIs really make me appreciate how far API design has come on 😂

ChaosCoder replied:
I was able to get rid of one level, as `usedAddressSize` can be passed via `&usedAddressSize`.

KaneCheshire commented on Tests/PeasyTests/ServerTests.swift:
```diff
@@ -0,0 +1,63 @@
+
+final class ServerTests: XCTestCase {
+
+    func testBindPort() throws {
```
Nice tests, thanks!

KaneCheshire commented:
Not sure why it's complaining about checks not being completed 🤔

ChaosCoder commented:
I refactored and addressed all your comments. The checks never ran at all, I also saw them never running on the other PR #20. Are you sure the CI is working?

KaneCheshire commented:
Yep CI ran for my last PR that I created on this repo, it's using GitHub actions and I don't know how that works across forks 

https://github.com/KaneCheshire/Peasy/pull/26/commits/4d5e56d0906725cf3e0fd4e443fc65649008c3cc
